tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,209 @@
+    fail::remove(DISK_FULL_LEADER);
+}
+
+fn fail_follower_full(cluster: &mut Cluster<ServerCluster>) {
+    std::thread::sleep(Duration::from_secs(2));
+    let key = String::from("2222").into_bytes();
+    let value = String::from("22220000").into_bytes();
+    let region = cluster.get_region(&key);
+    let target_peer = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 3)
+        .unwrap();
+    let leader = cluster.leader_of_region(region.get_id()).unwrap();
+    if leader.get_store_id() != target_peer.get_store_id() {
+        cluster.must_transfer_leader(region.get_id(), (*target_peer).clone());
+    }
+    let leader = cluster.leader_of_region(region.get_id()).unwrap();
+    let region = cluster.get_region(&key);
+    let follower1 = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 1)
+        .unwrap();
+    let follower2 = region
+        .get_peers()
+        .iter()
+        .find(|x| x.get_store_id() == 2)
+        .unwrap();
+
+    fail::cfg(DISK_FULL_FOLLOWER_1, "return").unwrap();
+    assert!(leader.get_store_id() == 3);
+    assert!(follower1.get_store_id() == 1);
+    assert!(follower2.get_store_id() == 2);
+    {
+        cluster.must_put(&key, &value);
+        let leader_state = cluster.raft_local_state(region.get_id(), leader.get_store_id());
+        let follower_state_1 = cluster.raft_local_state(region.get_id(), follower1.get_store_id());
+        let follower_state_2 = cluster.raft_local_state(region.get_id(), follower2.get_store_id());
+        assert!(leader_state.get_last_index() == follower_state_2.get_last_index());
+        assert!(leader_state.get_last_index() != follower_state_1.get_last_index());
+    }
+    {
+        //cluster.must_transfer_leader(region.get_id(), (*follower1).clone());
+        let timer = Instant::now();
+        loop {
+            cluster.reset_leader_of_region(region.get_id());
+            let cur_leader = cluster.leader_of_region(region.get_id());
+            if let Some(ref cur_leader) = cur_leader {
+                assert!(
+                    !(cur_leader.get_id() == follower1.get_id()
+                        && cur_leader.get_store_id() == follower1.get_store_id())
+                );
+            }
+            if timer.elapsed() > Duration::from_secs(2) {
+                //println!("transfer to followers with disk full will fail");
+                break;
+            }
+            cluster.transfer_leader(region.get_id(), (*follower1).clone());
+        }
+    }
+    fail::remove(DISK_FULL_FOLLOWER_1);
+}
+
+#[test]
+fn test_disk_full() {
+    let reserve = 1024 * 1024 * 100;
+    let mut cluster = new_server_cluster(0, 3);
+    cluster.cfg.storage.reserve_space = ReadableSize(reserve);
+    cluster.cfg.raft_store.pd_store_heartbeat_tick_interval =
+        ReadableDuration(Duration::from_secs(3000)); //disable disk status update influence.
+    cluster.run();
```

Comment:
`run` will bootstrap a region with id `1`, and peers (1, 1), (2, 2) and `(3, 3)` in the cluster. You can use them directly instead of using `region.get_peers().find(...)`.

Reply:
ok